tendrl/node_agent/message/handler.py

```diff
@@ -0,0 +1,53 @@
+import _socket
+from _socket import error as _socket_error
+import gevent.event
+import gevent.greenlet
+from gevent.server import StreamServer
+from io import BlockingIOError
+import os
+from tendrl.commons.message import Message
+from tendrl.node_agent.message.logger import Logger
+
+
+RECEIVE_DATA_SIZE = 4096
+PATH = "/etc/tendrl/.node_agent_socket"
+
+
+class MessageHandler(gevent.greenlet.Greenlet):
+    def __init__(self):
+        super(MessageHandler, self).__init__()
+        self.server = StreamServer(
+            self.bind_unix_listener(),
+            self.read_socket
+        )
+
+    def read_socket(self, sock, address):
+        try:
+            data = sock.recv(RECEIVE_DATA_SIZE)
+            message = Message.from_json(data)
+            Logger(message)
+        except (TypeError, ValueError, KeyError,
+                _socket.error, _socket.timeout) as ex:
+            raise ex
+
+    def _run(self):
+        try:
+            self.server.serve_forever()
+        except (TypeError, BlockingIOError, _socket_error, ValueError) as ex:
+            raise ex
+
+    def stop(self):
+        self.sock.close()
+        self.server.close()
+
+    def bind_unix_listener(self):
+        try:
+            if os.path.exists(PATH):
+                os.remove(PATH)
```

Comment:
@anmolbabu If we are not removed this file then i will throw exception like "Address already in use". They have to remove manually to run then node agent again.